### PR TITLE
feat: add csrf middleware

### PR DESCRIPTION
**Pull request by L2ncE**
#### What type of PR is this?

feat

#### What this PR does / why we need it (en: English/zh: Chinese):

en: add csrf middleware
zh: 新增 csrf 中间件

#### Which issue(s) this PR fixes:

https://github.com/cloudwego/hertz/issues/300

### CODE REVIEW

File: option.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+func WithSecret(secret string) Option {
+	return Option{
+		F: func(o *Options) {
+			o.Secret = secret
+		},
+	}
+}
+
```
Comment (li-jin-gou) [inline option.go:134]: add comment  and unit test

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module github.com/hertz-contrib/csrf
+
+go 1.18
+
+require (
+	github.com/cloudwego/hertz v0.3.2
```
Comment (li-jin-gou) [inline go.mod]: update hertz-version

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,55 +0,0 @@
-# How to Contribute
```
Comment (Duslia) [inline CONTRIBUTING.md:1]: There is no need to delete this file

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -1,128 +0,0 @@
-# Contributor Covenant Code of Conduct
```
Comment (Duslia) [inline CODE_OF_CONDUCT.md:1]: There is no need to delete this file

File: go.mod (hunk 2 above)
Comment (li-jin-gou) [reply go.mod:6]: It was aligned with other extension libraries, so it was removed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (li-jin-gou) [reply CONTRIBUTING.md:1]: It was aligned with other extension libraries, so it was removed.

File: README.md [hunk 5]
```diff
@@ -1 +1,69 @@
-# .github
\ No newline at end of file
+# CSRF (This is a community driven project)
+
+CSRF for hertz.
```
Comment (GuangmingLuo) [inline README.md]: What is CSRF and what does it used for ? - The least information you should mention here

File: go.mod (hunk 2 above)
Comment (L2ncE) [reply go.mod]: > update hertz-version

done

File: README.md (hunk 5 above)
Comment (L2ncE) [reply README.md]: > What is CSRF and what does it used for ? - The least information you should mention here

done

File: option.go (hunk 1 above)
Comment (L2ncE) [reply option.go:134]: > add comment and unit test

Added comments, other than that I think the unit tests are complete enough that every option is tested.

Comment (li-jin-gou) [reply option.go:134]: ok

**Review by li-jin-gou — APPROVED** [review]

**Review by justlorain — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**